[MM-30444][MM-23459][MM-31775] - Style sidebar navigator and team icons

nevyangelova opened the pull request:
#### Summary
This PR handles styling of the sidebar channel navigator, history arrows and filters. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30444
https://mattermost.atlassian.net/browse/MM-23459
https://mattermost.atlassian.net/browse/MM-31775
#### Screenshots
desktop
<img width="248" alt="Screenshot 2020-12-04 at 20 57 57" src="https://user-images.githubusercontent.com/16692883/101209121-8c6c9f80-3673-11eb-9287-b79d6f802688.png">

webapp
<img width="252" alt="Screenshot 2020-12-04 at 20 58 42" src="https://user-images.githubusercontent.com/16692883/101209125-8d9dcc80-3673-11eb-8901-d899f9406991.png">

matthewbirtch commented on i18n/en.json:
```diff
@@ -3722,7 +3722,7 @@
   "sidebar_left.channel_navigator.goBackLabel": "Back",
   "sidebar_left.channel_navigator.goForwardLabel": "Forward",
-  "sidebar_left.channel_navigator.jumpTo": "Jump to...",
+  "sidebar_left.channel_navigator.jumpTo": "Find a channel...",
```
Let's change to "Find channel" - remove the 'a' and the '...' - it doesn't fit well when you hover and the shortcut shows

matthewbirtch reviewed (commented):
Thanks @nevyangelova. Nice work on this. A few minor updates I have. I tried to add comments inline on the css. I hope I caught all the places to update. I could only do this for the webapp layout since I can't inspect on the desktop. 

### Both layouts:
- Can we change the padding on the `SidebarChannelNavigator_jumpToButton`?
```
padding: 0 2px;
```
- The icons and content within the buttons seem slightly off-centered vertically

![image](https://user-images.githubusercontent.com/2040554/101429966-174ad580-38d2-11eb-85c6-8f9db6b72db6.png)

- Let's change 'Find a channel…' to just 'Find channel' since it doesn't really fit in the webapp layout when you hover
- We'll then want to adjust the position of the shortcut as well
![image](https://user-images.githubusercontent.com/2040554/101428190-3e070d00-38ce-11eb-97fc-1e5e25a20843.png)


### Desktop app (two-line layout)
- The divider line between the filter and history arrows should be 1px wide. It looks 2px wide
![image](https://user-images.githubusercontent.com/2040554/101429243-7871a980-38d0-11eb-9014-5281fba4a14d.png)
- For this layout, can we have all corners rounded with the same rounded radius, It looks a bit odd with the arrow button hover states
![image](https://user-images.githubusercontent.com/2040554/101429421-eb7b2000-38d0-11eb-94d7-6532651e30a0.png)
- Also, for this layout, we need to adjust the hover state to match the other `+` icon to be 8% opacity rather than 24% since it doesn't have the background color on the default state in this layout.

Let me know if you'd like to chat through any of this.

nevyangelova commented:
Thanks @matthewbirtch resolved! Can you please advice on the team icons as well? Particularly the drag and drop states, which currently have the side handle when dragging. This wasn't reflected in the design. IMO we don't need those drag handles especially if we are adding an all round border. Let me know :)

matthewbirtch commented:
Thanks @nevyangelova. 

Looks like a few more things
1. Both layouts: the 'Find channel' button hover state didn't need to change it's opacity since it's contained in a shape in the default state
2. Desktop app layout: filter button hover state needs to match the updates you made for the arrows and plus button:
![image](https://user-images.githubusercontent.com/2040554/101496920-57dd3a00-3938-11eb-93d3-03edea5ccd01.png)
3. Webapp layout: hover states for the filter, find channel, and plus button didn't need to change. The opacity change I recommended in the last review was only meant for the desktop app because those buttons are not contained in a shape in their default state in the desktop layout
![image](https://user-images.githubusercontent.com/2040554/101497364-e2259e00-3938-11eb-8a90-40e4d9eada87.png)

> Can you please advice on the team icons as well? Particularly the drag and drop states, which currently have the side handle when dragging. This wasn't reflected in the design. IMO we don't need those drag handles especially if we are adding an all round border. Let me know :)
Can you move the team icons work to a separate PR? There is a separate Jira Ticket (https://mattermost.atlassian.net/browse/MM-23459) that handles this and Mike Gamble did the design work on it, so it would be best for him to review.

nevyangelova commented:
Oh good to know! I will add him a a reviewer once I go over the design! It makes sense to keep these together as its related styling changes.

nevyangelova commented:
@michaelgamble can you provide guidance on the styling of the team icons?

esethna commented:
@nevyangelova I agree with Matt it may make sense to pull the team icon work into a different PR. It is likely going to need a few rounds of iteration to get the animations and styling just right, there's a lot of detail to it in the ticket https://mattermost.atlassian.net/browse/MM-23459 (so we don't want to block the channel sidebar work on it).

nevyangelova commented:
@esethna ok that makes sense. I am meeting Michael on Monday and we’re going to try to adjust everything via zoom and if it’s not done by Monday I will split into two PRs

hmhealey commented on components/sidebar/channel_navigator/channel_navigator.tsx:
```diff
@@ -53,47 +60,79 @@ export default class ChannelNavigator extends React.PureComponent<Props, State>
+            />
+        );
+
+        const historyArrows = (
```
Minor nit, can you move this into the desktop app branch of the if statement below?

hmhealey reviewed (changes requested):
I think there's still style tweaks to made, but I have a thought about code organization I want to mention.

Also, I propose we should follow the name in the code and call those controls in the sidebar the ChannelNavigator to avoid confusion around that 😛

matthewbirtch reviewed (changes requested):
@nevyangelova the interactions on the team sidebar seem like they need a bit more polish to me. 
1. Animation is pretty jittery when my mouse is near the edges of the buttons
2. The hover animation doesn't feel buttery smooth - it seems like the team icons are shifting position a bit on hover. I see a `margin:auto` on hover that might be causing this. When I remove that css rule, it seems to fix the issue
3. The plus button hover state is most noticeable as the icon jumps around on hover. See the screen capture below...

![2020-12-18 09 47 52](https://user-images.githubusercontent.com/2040554/102627476-3e957400-4116-11eb-8cdb-71807cba6366.gif)

matthewbirtch commented:
@michaelgamble I'll let you do the ux review of the team icons portion, but I wonder if we should transform the scale on click rather than transforming the y position? Something like
`transform: scale(0.92);`

matthewbirtch reviewed (changes requested):
@nevyangelova did something get changed on the desktop layout? It doesn't look quite right anymore. Notice the color of the 'Find Channel' button and the position of the shortcut

**Implemented:**
![image](https://user-images.githubusercontent.com/2040554/103816143-841cd100-5032-11eb-9c34-d3dbf80d3a6d.png)


**Designed:**
![image](https://user-images.githubusercontent.com/2040554/103816101-736c5b00-5032-11eb-9da2-8bca94180a12.png)

The hover states on the filter/history and plus controls needs to be updated on desktop as well. The history arrow button in the example below shows how it should be on hover

![image](https://user-images.githubusercontent.com/2040554/103816349-d6f68880-5032-11eb-865a-18fd0f31dba1.png)

Have a look at this prototype: https://www.figma.com/proto/zVNYaShAXDK8i1RdpuEkr8/Channel-Sidebar-All-Phases?node-id=2728%3A46345&viewport=-156%2C339%2C0.14608465135097504&scaling=min-zoom

nevyangelova commented:
Sorry @matthewbirtch some of the hover states got lost between the commits. Fixed now. @michaelgamble implemented the final notes for the team icons as well.

matthewbirtch commented:
Thanks @nevyangelova. It looks like you've resolved the issues for the desktop app layout, but there are some outstanding issues for the browser layout.

1. Background color for the **default** state of the filter/find channels/plus buttons should have an opacity of 8%. Currently it shows at 16%
Implemented:
![image](https://user-images.githubusercontent.com/2040554/104064757-7ac76c00-51cc-11eb-86e2-32bbe4c5fedf.png)
Designed:
![image](https://user-images.githubusercontent.com/2040554/104064782-861a9780-51cc-11eb-8a6a-fc838532628b.png)

2. The background color for the **hover** state for these buttons should be 16%. Currently the hover state only changes the color of the icons and text
Implemented:
![image](https://user-images.githubusercontent.com/2040554/104064816-97fc3a80-51cc-11eb-832f-a47b010896a2.png)
Designed:
![image](https://user-images.githubusercontent.com/2040554/104064860-ab0f0a80-51cc-11eb-806e-c44cf5a1a5ac.png)

For the team icons, things are looking great. I just had one question:
Should the team icons with images also have the dark blue outline between the light blue outline and the image like the text-based icons do?
![2021-01-08 16 15 55](https://user-images.githubusercontent.com/2040554/104064961-e3164d80-51cc-11eb-934f-b58dac697d94.gif)

nevyangelova commented:
@matthewbirtch fixed. To your third question - per design it looks like the borders will be the same as no image icons https://www.figma.com/proto/s03qVXlOQWh1YWWMkSVonw/Team-Icon-Interactions?node-id=202%3A33526&scaling=min-zoom

matthewbirtch commented:
Looks like the layout for the filter/search/plus buttons are good to go now. 

> @matthewbirtch fixed. To your third question - per design it looks like the borders will be the same as no image icons https://www.figma.com/proto/s03qVXlOQWh1YWWMkSVonw/Team-Icon-Interactions?node-id=202%3A33526&scaling=min-zoom

Yep, you're right. So, are you able to update this so that the dark blue outline shows between the image and the light blue border?
![image](https://user-images.githubusercontent.com/2040554/104212637-6de08d80-5403-11eb-9540-36f4c97e4ea1.png)

nevyangelova commented:
@matthewbirtch I had a video chat with @michaelgamble and we discussed that when an image is uploaded we don't want the internal shadow but maybe I misunderstood how it should look. Michael can provide guidance.

michaelgamble commented:
@nevyangelova I think that last point regarding the internal shadow was a miscommunication, it wasnt that I wanted it removed, but in its state during our review, it was visually incorrect and needed to be fixed to match the design (it had some back color bleeding through the edges of where the corner radius's were showing)...

I think that was my miscommunication in not being clear enough in what i was seeing as the issue.

matthewbirtch reviewed (approved):
Approving this for the channel sidebar navigator updates. Leaving it to @michaelgamble to approve for the team icons interactions

hmhealey commented on components/widgets/team_icon/team_icon.tsx:
```diff
@@ -36,34 +40,38 @@ export class TeamIcon extends React.PureComponent<Props> {
         }
         return (
-            <div className={`TeamIcon ${hoverCss} TeamIcon__${size}`} >
+            <div className={classNames(`TeamIcon ${hoverCss} TeamIcon__${size} ${className}`, {withImage: teamIconUrl})}>
```
Can you move the `className` out of the template string? It's adding `undefined` to the CSS class if you look at the snapshot

michaelgamble reviewed (changes requested):
A couple of changes needed:
- the inner stroke color needs to respect theme color variables (it stays blue when you switch to dark theme)
- when an image exists for a team icon, on click it seems to scale the image smaller, when it really should just crop its bounds in
- Team icons without an image, the two letter text seems to jitter left and right on click (thats back)
- While dragging a team, the center opacity of a team with no image shifts to a less opacity, it should stay the same opacity as the hover state
- I noticed when i was changing the themes that the save button has some wierdness happening to it that doesn't exist currently on community daily, so im wondering if thier is some CSS bleed happening

nevyangelova commented:
> A couple of changes needed:
> 
> * the inner stroke color needs to respect theme color variables (it stays blue when you switch to dark theme)
> * when an image exists for a team icon, on click it seems to scale the image smaller, when it really should just crop its bounds in
> * Team icons without an image, the two letter text seems to jitter left and right on click (thats back)
> * While dragging a team, the center opacity of a team with no image shifts to a less opacity, it should stay the same opacity as the hover state
> * I noticed when i was changing the themes that the save button has some wierdness happening to it that doesn't exist currently on community daily, so im wondering if thier is some CSS bleed happening

@michaelgamble 
1. Fixed - however we don't have a colour variable for the team sidebar. There is another div that uses the Sidebar header colour as a background and the team sidebar applies a shadow on top. It would be possible to add a theme variable that matches but it will be a big change and it will require a separate PR, possibly won't make GA.
2. Fixed
3. I don't see this issue its fixed for me
4. Fixed
5. If this is on community daily its need a bug ticket, would you like me to file one?

esethna commented:
@michaelgamble please help with reviewing and approving as we are aiming to merge this asap

michaelgamble reviewed (changes requested):
> > A couple of changes needed:
> > 
> > * the inner stroke color needs to respect theme color variables (it stays blue when you switch to dark theme)
> > * when an image exists for a team icon, on click it seems to scale the image smaller, when it really should just crop its bounds in
> > * Team icons without an image, the two letter text seems to jitter left and right on click (thats back)
> > * While dragging a team, the center opacity of a team with no image shifts to a less opacity, it should stay the same opacity as the hover state
> > * I noticed when i was changing the themes that the save button has some wierdness happening to it that doesn't exist currently on community daily, so im wondering if thier is some CSS bleed happening
> 
> @michaelgamble
> 
> 1. Fixed - however we don't have a colour variable for the team sidebar. There is another div that uses the Sidebar header colour as a background and the team sidebar applies a shadow on top. It would be possible to add a theme variable that matches but it will be a big change and it will require a separate PR, possibly won't make GA.
> 2. Fixed
> 3. I don't see this issue its fixed for me
> 4. Fixed
> 5. If this is on community daily its need a bug ticket, would you like me to file one?

1. We need to figure out a solution to get the correct color
2. The scaling issue is fixed.
3. This issue is still happening, see the attached GIF, my environment is BigSur v11.1 chrome (latest version)
![2021-01-14 14 38 33](https://user-images.githubusercontent.com/31866506/104641168-6de8c380-5677-11eb-9449-eed3c9fdee0b.gif)
4. The opacity while dragging issue is fixed.
5. We should still look into this, i was saying this issue is prevalent on our test instance here but not prevalent on community-daily so its likely a bug we are introducing.(maybe?)

A new issue:
- While dragging a team icon with an image, the stroke style changes, this shouldnt happen it should remain consistent while dragging (team icons without images is working correctly)

hmhealey commented:
- For 1, The description for https://mattermost.atlassian.net/browse/MM-23459 says that it should use the "Sidebar Active" colour, but that's a bit ambiguous. It looks to me like it's using Sidebar Text Active Border (the little bit of colour on the left of the selected team, but should it instead use the Sidebar Text Active Color (which would be white or light grey in most cases)?
- For 3, I see this specifically on Chrome, but not on Firefox, so that might be why it's no reproing consistently. Usually it's Firefox that screws up the styling, but I guess not in this case.
- For 5, can you file a ticket for that @michaelgamble? I'm not sure what you're seeing, but there's nothing obviously wrong to me there. I can't find any changes that would affect that modal in this PR though.

nevyangelova commented:
@michaelgamble @hmhealey 
1. Fixed with a new element casting shadow on top of our borders, however I don't think this is a good solution because it layers unnecessary elements and I think adding a theme variable will be the best option for this. Also this element cannot be animated as part of our sequence and won't be visible when the icon is dragging. 

3. I still can't repro on any browser. Maybe I can update my OS over the weekend and investigate. I would open a ticket

New issue: fixed

For additional CSS problems which are not related to Channel navigator or team icons please file bug tickets.

asaadmahmood commented:
@nevyangelova We already have a lot of variables, adding theme variables is against our long term goals. We can still animate it based on the active or hover or any other attributes right by targeting the pseudo class?

michaelgamble reviewed (approved):
Looks like weve got it 👍

michaelgamble commented:
> @nevyangelova We already have a lot of variables, adding theme variables is against our long term goals. We can still animate it based on the active or hover or any other attributes right by targeting the pseudo class?

We brought this up in our design meeting today and came to a different conclusion about the long term goals (ultimately supporting the decision to have a team sidebar color, but we will table that discussion for the core efforts around the design system.

jgilliam17 commented:
/update-branch

jgilliam17 commented:
@hmhealey 
Top menu has some unexpected styling, no hover in light theme and wrong background in dark
<img width="670" alt="Screen Shot 2021-01-15 at 4 29 30 PM" src="https://user-images.githubusercontent.com/52937121/104780784-511fbf00-574f-11eb-889c-16ab052bb08b.png">
<img width="732" alt="Screen Shot 2021-01-15 at 4 29 10 PM" src="https://user-images.githubusercontent.com/52937121/104780791-53821900-574f-11eb-85f7-b7cacc3ab5a5.png">

jgilliam17 commented:
@hmhealey One more thing
Team icons appear to jump on drop. See video.

https://user-images.githubusercontent.com/52937121/104782436-2b47e980-5752-11eb-8488-1c5eadd1617e.mov

nevyangelova commented:
@jgilliam17 this PR is for the icons and styling of the sidebar category any additional changes implemented by @hmhealey for the unreads category should be filed in a separate ticket. Can we keep the comments in the review relevant for the changes implemented here because I am not certain what to prioritise fixing.

jgilliam17 commented:
@nevyangelova 
Exp. sidebar fails to enable on the test server. It appears to save in System Console, but can't be toggled ON in Account Settings

hmhealey commented:
I'll go force it on. I think we ran into some timing issues because of some of the removal changes have been merged in.

hmhealey commented:
/update-master

jgilliam17 commented:
@nevyangelova 
Team icons no longer jumps on drop, looks good.
Top Menu styling looks good, including the hover in dark and light themes.
Few UX issue 
1. Filter (when Unreads are disabled) is missing background color/shading
<img width="449" alt="Screen Shot 2021-01-18 at 4 43 09 PM" src="https://user-images.githubusercontent.com/52937121/104966213-5464b600-59ae-11eb-8be9-37742fb81dd6.png">

2. History icons/arrows style is wrong on desktop 
<img width="425" alt="Screen Shot 2021-01-18 at 4 51 26 PM" src="https://user-images.githubusercontent.com/52937121/104966324-9e4d9c00-59ae-11eb-99af-7ed7d989c818.png">

nevyangelova commented:
Point 1 is how UX wanted it, so it will match the other buttons (see screenshots https://mattermost.atlassian.net/browse/MM-30444). Will fix the other issue.

jgilliam17 reviewed (approved):
Thank you @nevyangelova 
Tested, looks good to merge.
- Verified styling of the sidebar channel navigator, history arrows and filters in browsers and desktop app. 
- Verified Team icon UI improvement 
- Verified when Group Unread setting is ON,  unread filter is hidden.
Adding history arrow alignment issue for follow up
https://mattermost.atlassian.net/browse/MM-32138
